### PR TITLE
EREGCSC-1458 Resource Download Button

### PR DESCRIPTION
**Pull request by thwalker6**
Resolves #

1458
**Description-**

Creates a download all resources button for the resources page

**This pull request changes...**

- expected change 1
A button wil appear on the resources page to download all resources in a csv

1. steps to view and verify change
Go to the search resources page.

Search for resources with filter and or a query or neither

Hit the button.

All resources will be downloaded in a csv that are searched for.

### CODE REVIEW

File: solution/ui/regulations/eregs-vite/src/components/resources/ResourceExportBtn.vue [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                responseContent = responseContent.concat(content.results)
+
+            }
+            return responseContent
+        },
+    },
+}
+</script>
+<style scoped>
+.csv-button {
```
Comment (PhilR8) [inline solution/ui/regulations/eregs-vite/src/components/resources/ResourceExportBtn.vue:104]: Is this `csv-button` class being used anywhere?

File: solution/ui/regulations/eregs-vite/src/components/resources/ResourceExportBtn.vue [hunk 2]
```diff
@@ -0,0 +1,121 @@
+<template>
+    <div class="resource_btn_container">
+        <button class="default-btn action-btn search_resource_btn" @click="createCSV">
```
Comment (PhilR8) [inline solution/ui/regulations/eregs-vite/src/components/resources/ResourceExportBtn.vue:3]: the button needs to be the same height as the sort selectbox.  Right now it's too tall: 
Screen Shot 2022-09-09 at 16 05 48

changing the `action-btn` top and bottom padding to `8px` should do the trick.

Comment (PhilR8) [reply solution/ui/regulations/eregs-vite/src/components/resources/ResourceExportBtn.vue:3]: we also need to handle narrow screen widths.  During planning we'd talked about just changing the button label to be "CSV" like this:

Screen Shot 2022-09-09 at 16 09 56

We should also add an `aria-label` to this button with something like "Download Spreadsheet as CSV" when the label is shortened.

File: solution/ui/regulations/eregs-vite/src/components/resources/ResourceExportBtn.vue [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        categories: {
+            type: Array,
+            default:() => []
+        },
+        supCount: {
+            type: Number,
+            default: 0
+        }
+    },
+    data: {
```
Comment (PhilR8) [inline solution/ui/regulations/eregs-vite/src/components/resources/ResourceExportBtn.vue:42]: data needs to be a function with an explicit return.  You can use the method shorthand and write it like 

```
    data() {
        return {
            supplementalContent: []
        }
    },
```

File: solution/ui/regulations/eregs-vite/src/components/resources/ResourceExportBtn.vue [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            }
+            return locString
+        },
+        async getSupplementalContent() {
+            let supNum = 0;
+            let page = 1;
+            let content = []
+            let responseContent = []
+            while (supNum < this.supCount) {
+                content = getSupplementalContentV3({
```
Comment (PhilR8) [inline solution/ui/regulations/eregs-vite/src/components/resources/ResourceExportBtn.vue:86]: this is not currently working locally or on the experimental deployment

**Review by PhilR8 — CHANGES_REQUESTED** [review]
The download functionality isn't working locally or on the experimental deployment.

I've also added some minor styling and breakpoint issues to be addressed.

File: solution/ui/regulations/eregs-vite/src/components/resources/ResourceExportBtn.vue (hunk 2 above)
Comment (PhilR8) [reply solution/ui/regulations/eregs-vite/src/components/resources/ResourceExportBtn.vue:3]: We should also add an `aria-label` to this button with something like "Download Spreadsheet as CSV" when the label is shortened.

File: solution/ui/regulations/eregs-vite/src/components/resources/ResourceExportBtn.vue (hunk 1 above)
Comment (thwalker6) [reply solution/ui/regulations/eregs-vite/src/components/resources/ResourceExportBtn.vue:104]: removed

File: solution/ui/regulations/eregs-vite/src/components/resources/ResourceExportBtn.vue (hunk 3 above)
Comment (thwalker6) [reply solution/ui/regulations/eregs-vite/src/components/resources/ResourceExportBtn.vue:42]: changed

File: solution/ui/regulations/eregs-vite/src/components/resources/ResourceExportBtn.vue (hunk 4 above)
Comment (thwalker6) [reply solution/ui/regulations/eregs-vite/src/components/resources/ResourceExportBtn.vue:86]: fixed

**Review by PhilR8 — APPROVED** [review]
These changes look good and the download functionality appears to work well!  Nice work.  LGTM!

**Pull request merged**